biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -259,8 +259,20 @@ public void analyze() throws Exception {
 				Set<Instruction> unused = Create.set();
 
 				Instructions filter = new Instructions(getExportPackage());
-				filter.append(getExportContents());
-				filter.append(getExportedByAnnotation());
+				filter.appendIfAbsent(getExportContents());
+
+				//
+				// get the packages exported by the annotation. However,
+				// we remove any explicitly explicitly named private package so
+				// that
+				// you can always override an annotation
+				//
+
+				Parameters exportedByAnnotation = getExportedByAnnotation();
+				exportedByAnnotation.keySet()
+					.removeAll(getPrivatePackage().keySet());
```

Comment:
I think this is wrong. You need to merge since the Export annotation can contain important information like `uses`, `substitution` as well as attributes and directives.

Reply:
Nope, see the test case. bnd already merges (and properly prioritizes) the attributes from package info like version. I actually considered to remove this because sometime you need to override this but found this too much work. In that rare case you can override the info on the Export-Package. I think you can also remove attrs/dirs.